wcfsetup/install/files/lib/system/service/worker/ServiceWorkerHandler.class.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace wcf\system\service\worker;
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Psr7\Request;
+use Jose\Component\KeyManagement\JWKFactory;
+use ParagonIE\ConstantTime\Base64UrlSafe;
+use ParagonIE\ConstantTime\Binary;
+use wcf\data\option\OptionEditor;
+use wcf\data\service\worker\ServiceWorker;
+use wcf\system\io\HttpFactory;
+use wcf\system\SingletonFactory;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class ServiceWorkerHandler extends SingletonFactory
+{
+    /**
+     * Maximum payload length that can be sent to the service worker.
+     * @see https://stackoverflow.com/a/66222350
+     */
+    public const MAX_PAYLOAD_LENGTH = 2847;
+
+    /**
+     * Time to live for the notification before it is discarded by the push-service provider.
+     */
+    public const TTL = 604800; // 7 days
+
+    private ClientInterface $client;
+
+    /**
+     * @internal
+     * @TODO
+     */
+    public static function createNewKeys(): void
+    {
+        $jwk = JWKFactory::createECKey(Encryption::CURVE_ALGORITHM);
+        $binaryPublicKey = \hex2bin(Util::serializePublicKey($jwk->get('x'), $jwk->get('y')));
+        $binaryPrivateKey = \hex2bin(
+            \str_pad(
+                \bin2hex(Base64UrlSafe::decode($jwk->get('d'), true)),
+                2 * VAPID::PRIVATE_KEY_LENGTH,
+                '0',
+                STR_PAD_LEFT
+            )
+        );
+        OptionEditor::import([
+            'service_worker_public_key' => Base64UrlSafe::encodeUnpadded($binaryPublicKey),
+            'service_worker_private_key' => Base64UrlSafe::encodeUnpadded($binaryPrivateKey),
+        ]);
+    }
+
+    /**
+     * Send the given payload to the service worker.
+     *
+     * @param ServiceWorker $serviceWorker
+     * @param string $payload
+     */
+    public function sendToServiceWorker(ServiceWorker $serviceWorker, string $payload): void
+    {
+        if (Binary::safeStrlen($payload) > self::MAX_PAYLOAD_LENGTH) {
```

Comment:
```suggestion
        if (\strlen($payload) > self::MAX_PAYLOAD_LENGTH) {
```
(same below, see the note in the other comment for the reasoning)